### ISSUE TITLE
client commands run by tasks: detect ssh comms

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#2214]
Even if a job host needs `task communications method = ssh`, general client commands run by users on that job host need to use the `--use-ssh` option to stop them from defaulting to HTTPS comms (or Pyro pre-cylc-7).  Client commands in task job scripts, however, should detect `$CYLC_TASK_COMMS_METHOD=ssh` and automatically use `--use-ssh`.,

**Comment by hjoliver** [comment]
Example, currently in a suite using `task communications method = ssh` to a particular job host, on that job host if a task (whose messaging commands automatically use ssh) tries to broadcast settings back to the suite, it must explicitly use the ssh and host options:

```
cylc broadcast --use-ssh --host=HOST <normal broadcast args and options...> SUITE
```

Does this suggest the comms method setting should really be in the suite contact file  - so that even user- (not task-)invoked client commands executed on a job host can automatically determine the host and port and comms method from the contact file (meaning a different contact file for different hosts, unfortunately)

**Comment by hjoliver** [comment]
This sort of thing should really be documented too.

**Comment by matthewrmshin** [comment]
One thing I have considered (while doing #2141) was to add `CYLC_TASK_COMMS_METHOD=ssh` to `.service/contact` for hosts that apply. (It was not done because I did not want to complicate #2141, e.g. it is not yet clear to me how to handle this if the job hosts have shared file system with the suite host.)

**Comment by hjoliver** [comment]
> ... if the job hosts have shared filesystem

Yeah, tricky.  Maybe a single suite contract file with host entries in it, something like this:
```
CYLC_TASK_COMMS_METHOD = https
CYLC_TASK_COMMS_METHOD:host1 = ssh
```

If a host-specific entry exists, it should override the default on that host.

**Comment by matthewrmshin** [comment]
(This is complicated by the fact that a suite host may know a job host by a different name, so when a client on `host1` looks at the contact file, it does not know that `host1` refers to itself.)

**Comment by matthewrmshin** [comment]
(A variant of the above complication is that while we use the login node as the platform to submit a job to a cluster, the job typically runs on a compute node, so we'll need a system that is independent of the host name of the job submit node.)

**Comment by matthewrmshin** [comment]
See https://github.com/cylc/cylc/issues/2463#issuecomment-340067978, i.e.:

We should move the SSH+HTTP(S) communication logic from the messaging module to the network API client module. We'll introduce a simple command on the suite host to handle any method in the API client.
* The command will be expected to be invoked via SSH (although not necessarily), perhaps with STDIN being used to accept the API method arguments encoded in JSON. (Note: not quite #2123!)
* In theory, this should allow even the GUI on a remote host to do SSH+HTTP(S) communication with a suite.
